src/base/types/derivedmodels/GeckoModel.jl

```diff
@@ -0,0 +1,423 @@
+
+Returns the genes (proteins) in the order as they appear as variables in the
+model.
+"""
+genes(model::GeckoModel) = model.gene_ids
+
+"""
+    n_genes(model::GeckoModel)
+
+Returns the number of genes in the model.
+"""
+n_genes(model::GeckoModel) = length(model.gene_ids)
+
+"""
+    metabolites(model::GeckoModel)
+
+Return the metabolites in `model`.
+"""
+metabolites(model::GeckoModel) = model.metabolites
+
+"""
+    n_metabolites(model::GeckoModel) = 
+
+Return the number of metabolites in `model`.
+"""
+n_metabolites(model::GeckoModel) = length(metabolites(model))
+
+"""
+    bounds(model::GeckoModel)
+
+Return variable bounds for `GeckoModel`.
+"""
+bounds(model::GeckoModel) = (model.xl, model.xu)
+
+"""
+    coupling(model::GeckoModel)
+
+Coupling constraint matrix for a `GeckoModel`.
+"""
+coupling(model::GeckoModel) = model.C
+
+"""
+    coupling_bounds(model::GeckoModel)
+
+Coupling bounds for a `GeckoModel`.
+"""
+coupling_bounds(model::GeckoModel) = (model.cl, model.cu)
+
+"""
+    reaction_flux(model::MetabolicModel)
+
+Helper function to get fluxes from optimization problem.
+"""
+function reaction_flux(model::GeckoModel)
+    R = spzeros(n_reactions(model), n_genes(model) + length(model.irrev_reaction_ids))
+    for (i, rid) in enumerate(reactions(model))
+        for_idx = findfirst(
+            x -> x == rid * "§ARM§FOR" || x == rid * "§FOR",
+            model.irrev_reaction_ids,
+        )
+        rev_idx = findfirst(
+            x -> x == rid * "§ARM§REV" || x == rid * "§REV",
+            model.irrev_reaction_ids,
+        )
+        !isnothing(for_idx) && (R[i, for_idx] = 1.0)
+        !isnothing(rev_idx) && (R[i, rev_idx] = -1.0)
+    end
+    return R'
+end
+
+"""
+    GeckoModel(
+        model::StandardModel;
+        rid_isozymes = Dict{String, Vector{Isozyme}}(),
+        enzyme_capacities = [(),],
+    ) 
+
+Construct a `GeckoModel` based on `model` using the kinetic data encoded by 
+`rid_isozymes`. Enzyme capacity constraints can be added through `enzyme_capacities`,
+which is a vector of tuples. In the first position of the tuple is a list of gene ids,
+and the second position is mass upperbound of the sum of these gene ids. 
+
+The units of the fluxes and protein concentration depend on those used in
+`rid_isozymes` for the kcats and the molar masses encoded in  the genes of
+`model`. Currently only `modifications` that change attributes of the
+`optimizer` are supported.
+
+# Example
+```
+gm = GeckoModel(
+    model;
+    rid_isozymes,
+    enzyme_capacities = [(get_genes_with_kcats(rid_isozymes), total_protein_mass)],
+)
+
+opt_model = flux_balance_analysis(
+    gm,
+    Tulip.Optimizer
+)
+
+rxn_fluxes = flux_dict(gm, opt_model)
+prot_concens = protein_dict(gm, opt_model)
+```
+"""
+function GeckoModel(
+    model::StandardModel;
+    rid_isozymes = Dict{String,Vector{Isozyme}}(),
+    enzyme_capacities = [()],
+)
+    S, lb_fluxes, ub_fluxes, reaction_map, metabolite_map =
```

Comment:
the code from this should be moved into a standalone function in `analysis/`, e.g. `make_gecko_model` or so. (Also corresponds to what's in `test/`